components/src/dynamo/common/tests/multimodal/test_image_loader.py

```
@@ -245,3 +245,90 @@ async def test_cache_is_lru_not_fifo(loader: ImageLoader) -> None:
+async def test_load_image_batch_uuid_only_returns_none_slot(
+    loader: ImageLoader,
+) -> None:
+    """UuidOnly slot must produce a None entry — vLLM's mm_processor_cache resolves it."""
+    items = [{"UuidOnly": "img-aabbccdd"}]
+    result = await loader.load_image_batch(items)
+    assert result == [None]
+
+
+async def test_load_image_batch_mixed_url_and_uuid_only(loader: ImageLoader) -> None:
+    """5 images: 1 fresh url, 4 uuid-only repeats — slot alignment preserved."""
+    mock_client = _mock_http_client()
```

Ruff (0.15.12)

[error] Undefined name `_mock_http_client`

(F821)